jerry-core/mem/mem-heap.c

```diff
@@ -243,46 +186,48 @@ void *mem_heap_alloc_block_internal (const size_t size)
   const size_t required_size = ((size + MEM_ALIGNMENT - 1) / MEM_ALIGNMENT) * MEM_ALIGNMENT;
   mem_heap_free_t *data_space_p = NULL;
 
-  VALGRIND_DEFINED_SPACE (&mem_heap.first, sizeof (mem_heap_free_t));
+  mem_heap_t *mem_heap = jerry_port_get_heap ();
```

Comment:
You make a good point, but hopefully we can come up with a different solution than adding more statics.

My goal over time is to remove as many statics as possible as I would like to run more than one instance of the engine at a time.

In this particular case, I think we can achieve similar performance by forcing `jerry_port_get_heap()` to be inlined. Would you be ok with that @zherczeg ?